### DISCUSSION TITLE
Usage Of databricks generated tables as an in/out in the Datapipeline Process

### DISCUSSION THREAD

**Discussion by BramhaAelem** [OpenLineage/OpenLineage#1494] [Ideas]
Hi,

First activity : Making HTTP Call to pull the lookup data and store it in datalake.
Second Activity : After the completion of first activity, invoking Azure databricks to use the lookup file and generate the output tables. What are all the steps should be follow to refer databricks generated tables facets as an output in the current activity & input to the subsequent activities in the pipeline.
When I configure spark generated tables as output to the current activity the existing spark metadata is not showing up. 
How can this be achievable.
Can someone please guide me on this. 
Note: After the execution of each activity in ADF Pipeline I am sending start and complete/fail event lineage to Marquez.

Thanks,
Brahma

**Comment by pawel-big-lebowski** [comment]
Hello @BramhaAelem, could you include Spark code sample that leads to the behaviour you're describing?

**Reply by BramhaAelem** [reply]
> Hello @BramhaAelem, could you include Spark code sample that leads to the behaviour you're describing?
PFA spark notebook.
[process_taxi_data.txt](https://github.com/OpenLineage/OpenLineage/files/10413553/process_taxi_data.txt)
In this notebook we are creating following tables.
Source Tables:
source_yellow_taxi.yellow_trips (This is generating in other notebook)
taxi_zone_lookup.csv (Created Temporary View On it)

Target Tables:
yellow_top10_dropoff
yellow_top10_pickup
yellow_trips_daily
yellow_trips_hourly

In Marquez dataset names are appearing as bellow. It has the metadata(facets) captured from spark execution plan.
Marquez Namespace: dbfs
Dataset:
/user/hive/warehouse/nyctlc.db/yellow_top10_dropoff
/user/hive/warehouse/nyctlc.db/yellow_top10_pickup
/user/hive/warehouse/nyctlc.db/yellow_trips_daily
/user/hive/warehouse/nyctlc.db/yellow_trips_hourly

<img width="958" alt="image" src="https://user-images.githubusercontent.com/10257186/212373932-eaf9344c-f6cb-4084-b6bd-9e248a758668.png">

When I configured above datasets as outputs in my Datapipeline  and posted an lineage event Refer the openlineage event which has been generated as per the api documentation. the above dataset facets are getting overwritten

{
        "eventType": "COMPLETE",
        "eventTime": "2023-01-20T16:44:04.7808119Z",
        "run": {
            "runId": "8c7fd631-bce9-44ec-934e-69c15620e7f8"
        },
        "job": {
            "name": "process_taxi_data",
            "namespace": "azure-openlineage-adf"
        },
        "outputs": [
            {
                "name": "/user/hive/warehouse/nyctlc.db/yellow_trips_daily",
                "namespace": "dbfs"
            },
            {
                "name": "/user/hive/warehouse/nyctlc.db/yellow_trips_hourly",
                "namespace": "dbfs"
            },
            {
                "name": "/user/hive/warehouse/nyctlc.db/yellow_top10_pickup",
                "namespace": "dbfs"
            },
            {
                "name": "/user/hive/warehouse/nyctlc.db/yellow_top10_dropoff",
                "namespace": "dbfs"
            }
        ],
        "producer": "https://github.com/OpenLineage/OpenLineage/blob/v1-0-0/client",
        "schemaURL": "https://openlineage.io/spec/1-0-5/OpenLineage.json"
    }

Kindly let me know if you need anymore details.

**Comment by pawel-big-lebowski** [comment]
Hi @BramhaAelem, I am still trying to understand the problem. Are you saying that running code: 
```
# MAGIC %sql
# MAGIC ---Trips based on hour Of day 
# MAGIC create or replace table nyctlc.yellow_trips_hourly using delta as 
# MAGIC select hour(tpepPickupDateTime) as hour_of_trip,count(*) as trip_cnt from source_yellow_taxi.yellow_trips 
# MAGIC group by hour(tpepPickupDateTime) order by trip_cnt desc
```
results in no metadata (like no schema) collected for `yellow_trips_hourly` table. 
I see there is exactly one activity related to this table in your code.

**Reply by BramhaAelem** [reply]
Hi @pawel-big-lebowski 

Spark SQL metadata is getting captured through the Spark Execution plan for each table. As depicted in the above image it's have the metadata(Schema fields) associated with it. When I have specify the spark generated datasets as an output in my openlineage event (which is copied in my earlier post) it's getting overwritten the metadata which has been generated by spark execution plan. 

please refer the below image for reference. 

<img width="960" alt="image" src="https://user-images.githubusercontent.com/10257186/213783943-444c7e36-5cd1-4230-b045-6937a5a085b2.png">

Thanks,
Brahma

**Comment by BramhaAelem** [comment]
hello @pawel-big-lebowski - Did you get a chance to look into the above and suggest your recommendations accordingly.
